snap/info.go

```diff
@@ -1077,18 +1077,11 @@ func InstanceName(snapName, instanceKey string) string {
 }
 
 // ByType sorts the given slice of snap info by types. The most
-// important types will come first. The "snapd" snap is handled
-// as well.
+// important types will come first.
 type ByType []*Info
 
 func (r ByType) Len() int      { return len(r) }
 func (r ByType) Swap(i, j int) { r[i], r[j] = r[j], r[i] }
 func (r ByType) Less(i, j int) bool {
-	if r[i].SideInfo.RealName == "snapd" {
-		return true
-	}
-	if r[j].SideInfo.RealName == "snapd" {
-		return false
-	}
-	return typeOrder[r[i].Type] < typeOrder[r[j].Type]
+	return r[i].Type.sortOrder() < r[j].Type.sortOrder()
```

Comment:
I'd capture this in a helper of Type so that the caller does not need to know if sorting order is low -> hi or hi -> low.

```
func (m Type) SortsBefore(other Type) bool {
    return m.sortOrder() < m.sortOrder()
}
```

Then the code here becomes:
```
func (r ByType) Less(i, j int) bool {
    return r[i].Type.SortsBefore(r[j].Type)
}
```